instrumentation/github.com/99designs/gqlgen/otelgqlgen/gqlgen.go

```diff
@@ -0,0 +1,151 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelgqlgen
+
+import (
+	"context"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/graphql/handler/extension"
+
+	otelcontrib "go.opentelemetry.io/contrib"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	oteltrace "go.opentelemetry.io/otel/trace"
+)
+
+const (
+	tracerName      = "go.opentelemetry.io/contrib/instrumentation/github.com/99designs/gqlgen/otelgqlgen"
+	extensionName   = "OpenTelemetry"
+	complexityLimit = "ComplexityLimit"
+)
+
+type Tracer struct {
+	serviceName             string
+	complexityExtensionName string
+	tracer                  oteltrace.Tracer
+}
+
+var _ interface {
+	graphql.HandlerExtension
+	graphql.ResponseInterceptor
+	graphql.FieldInterceptor
+} = Tracer{}
+
+func (a Tracer) ExtensionName() string {
+	return extensionName
+}
+
+func (a Tracer) Validate(_ graphql.ExecutableSchema) error {
+	return nil
+}
+
+func (a Tracer) InterceptResponse(ctx context.Context, next graphql.ResponseHandler) *graphql.Response {
+	ctx, span := a.tracer.Start(ctx, operationName(ctx))
+	defer span.End()
+	if !span.IsRecording() {
+		return next(ctx)
+	}
+
+	oc := graphql.GetOperationContext(ctx)
+	span.SetAttributes(
+		ServiceName(a.serviceName),
+		RequestQuery(oc.RawQuery),
+	)
+	complexityExtension := a.complexityExtensionName
+	if complexityExtension == "" {
+		complexityExtension = complexityLimit
+	}
+	complexityStats, ok := oc.Stats.GetExtension(complexityExtension).(*extension.ComplexityStats)
+	if !ok {
+		// complexity extension is not used
+		complexityStats = &extension.ComplexityStats{}
+	}
+
+	if complexityStats.ComplexityLimit > 0 {
+		span.SetAttributes(
+			RequestComplexityLimit(int64(complexityStats.ComplexityLimit)),
+			RequestOperationComplexity(int64(complexityStats.Complexity)),
+		)
+	}
```

Comment:
This feature is cool and I think it is worth adding a test for it to make sure it is always working.